src/e2e/app/cli/repo-tests/src/test_rest_api.rs

```diff
@@ -62,17 +92,49 @@ pub async fn test_rest_api_request_dataset_tail(kamu_api_server_client: KamuApiS
         )
         .await;
 
-    // 3. Get dataset tail
+    // 3. Try to get the dataset tail
     kamu_api_server_client
         .rest_api_call_assert(
+            None,
             Method::GET,
-            "empty-root-dataset/tail?limit=10",
+            "player-scores/tail?limit=10",
+            None,
+            StatusCode::NO_CONTENT,
             None,
+        )
+        .await;
+
+    // 4. Ingest data
+    kamu_api_server_client
+        .rest_api_call_assert(
+            Some(token),
+            Method::POST,
+            "player-scores/ingest",
+            Some(RequestBody::NDJSON(
+                indoc::indoc!(
+                    r#"
+                    {"match_time": "2000-01-01", "match_id": 1, "player_id": "Alice", "score": 100}
+                    {"match_time": "2000-01-01", "match_id": 1, "player_id": "Bob", "score": 80}
+                    "#,
+                )
+                .into(),
+            )),
             StatusCode::OK,
-            // TODO: add expected body
             None,
         )
         .await;
+
+    // 5. Get the dataset tail
+    kamu_api_server_client
+        .rest_api_call_assert(
+            None,
+            Method::GET,
+            "player-scores/tail?limit=10",
+            None,
+            StatusCode::OK,
+            Some(r#"{"data":[{"offset":0,"op":0,"system_time":"2024-06-18T15:00:00Z","match_time":"2000-01-01T00:00:00Z","match_id":1,"player_id":"Alice","score":100},{"offset":1,"op":0,"system_time":"2024-06-18T15:00:00Z","match_time":"2000-01-01T00:00:00Z","match_id":1,"player_id":"Bob","score":80}],"schema":"{\"fields\":[{\"name\":\"offset\",\"data_type\":\"Int64\",\"nullable\":true,\"dict_id\":0,\"dict_is_ordered\":false,\"metadata\":{}},{\"name\":\"op\",\"data_type\":\"Int32\",\"nullable\":false,\"dict_id\":0,\"dict_is_ordered\":false,\"metadata\":{}},{\"name\":\"system_time\",\"data_type\":{\"Timestamp\":[\"Millisecond\",\"UTC\"]},\"nullable\":false,\"dict_id\":0,\"dict_is_ordered\":false,\"metadata\":{}},{\"name\":\"match_time\",\"data_type\":{\"Timestamp\":[\"Millisecond\",\"UTC\"]},\"nullable\":true,\"dict_id\":0,\"dict_is_ordered\":false,\"metadata\":{}},{\"name\":\"match_id\",\"data_type\":\"Int64\",\"nullable\":true,\"dict_id\":0,\"dict_is_ordered\":false,\"metadata\":{}},{\"name\":\"player_id\",\"data_type\":\"Utf8\",\"nullable\":true,\"dict_id\":0,\"dict_is_ordered\":false,\"metadata\":{}},{\"name\":\"score\",\"data_type\":\"Int64\",\"nullable\":true,\"dict_id\":0,\"dict_is_ordered\":false,\"metadata\":{}}],\"metadata\":{}}"}"#),
```

Comment:
Such strings are very hard to read and maintain. Why not use a similar text vs `serde_json::Value` for `expected_response_body` so you could write the expected response using `json!` macro?

Consider also setting `"includeSchema": false` request parameter if you don't want to deal with this messy string. I imagine ensuring schema is exactly that is not the point of this test.

Reply:
> Consider also setting "includeSchema": false request parameter if you don't want to deal with this messy string. I imagine ensuring schema is exactly that is not the point of this test.

A most excellent comment -- grateful for it

Reply:
Done